src/qibo/transpiler/optimizer.py

```diff
@@ -70,150 +70,164 @@ def __call__(self, circuit: Circuit):
 
 
 class InverseCancellation(Optimizer):
-    """
-    Identifies and removes pairs of adjacent gates from a quantum circuit.
-    """
 
     def __init__(self):
-        self.inverse_gates = (
-            gates.H | gates.Y | gates.Z | gates.X | gates.CNOT | gates.CZ | gates.SWAP
-        )
+        """Identifies and removes pairs of adjacent gates from
+        a quantum circuit.
+        """
+
+        self.inverse_gates = gates.H|gates.Y|gates.Z|gates.X|gates.CNOT|gates.CZ|gates.SWAP
         self.pos_rm_inv_gate = []
 
     def __call__(self, circuit: Circuit) -> Circuit:
-        self.__find_pos_rm(circuit)
+        """Performs pattern recognition to detect and eliminate adjacent gate pairs
+        in the quantum circuit.
+
+        Args:
+            circuit (:class:`qibo.models.circuit.Circuit`): Circuit to have gates identified.
+
+        Returns:
+            circuit (:class:`qibo.models.circuit.Circuit`): A new circuit with the pairs of
+            adjacent gates removed.
+        """
 
-        if 0 == len(self.pos_rm_inv_gate) or circuit.gates_of_type(gates.FusedGate):
+        if (0 == circuit.ngates or circuit.gates_of_type(gates.FusedGate)):
             return circuit
 
-        tmp_new_circuit = circuit.copy(True)
+        self.__find_pos_rm(circuit.nqubits, circuit.queue)
+
+        if 0 == len(self.pos_rm_inv_gate):
+            return circuit
 
+        tmp_new_circuit = circuit.copy(True)
         new_circuit = circuit.__class__(**tmp_new_circuit.init_kwargs)
         for i, gate in enumerate(tmp_new_circuit.queue):
-            if not (i in self.pos_rm_inv_gate):
+            if not i in self.pos_rm_inv_gate:
                 new_circuit.add(gate)
-
         return new_circuit
 
-    def __find_pos_rm(self, newCircuit: Circuit):
-        """
-        Identifies and marks pairs of inverse gates that can be removed from the circuit.
+    def __find_pos_rm(self, number_qubits: int, list_gates: list):
```

Comment:
```suggestion
    def _find_pos_rm(self, number_qubits: int, list_gates: list):
```
One underscore is enough for custom hidden methods and functions.

Reply:
Ok, I changed it.